ateam_ai/src/plays/defense.hpp

```diff
@@ -0,0 +1,125 @@
+{
+  DirectedGraph<BehaviorGoal> defense_graph;
+  // Go to the middle of the goalie area
+  BehaviorGoal goalie = get_goalie_behavior_goal(our_side_info);
+  defense_graph.add_node(goalie);
+
+  std::vector<BehaviorGoal> defenders = get_defense_behavior_goals(world, field, 2);
+  for (BehaviorGoal defender : defenders) {
+    defense_graph.add_node(defender);
+  }
+  return defense_graph;
+}
+
+BehaviorGoal get_goalie_behavior_goal(const FieldSidedInfo & our_side_info)
+{
+  // Line that is 0.5 m from the defense area in all directions
+  ateam_geometry::Segment goalie_line = ateam_geometry::Segment(
+    ateam_geometry::Point(
+      -4,
+      0.5), ateam_geometry::Point(
+      -4, -0.5));
+
+  // Get the ball location
+  std::optional<Eigen::Vector2d> ball_location = world.get_unique_ball();
+  while (!ball_location.has_value()) {
+    ball_location = world.get_unique_ball();
+  }
+
+  // Get the point on the goalie line that is closest to the ball
+  ateam_geometry::Point _goalie_point = goalie_line.projection(ball_location);
+
+  // Have the goalie defend the goal by going to that point
+  return BehaviorGoal goalie {
+    BehaviorGoal::Type::MoveToPoint,
+    BehaviorGoal::Priority::Required,
+    MoveParam(_goalie_point)
+  };
+}
+
+std::vector<BehaviorGoal> get_defense_behavior_goals(
+  const World & world, const Field & field,
+  const int & num_defenders)
+{
+  std::vector<BehaviorGoal> defenders;
+  std::optional<Eigen::Vector2d> ball_location = world.get_unique_ball();
+  // Get line between the ball and the goal
+  while (!ball_location.has_value()) {
+    ball_location = world.get_unique_ball();
+  }
+  ateam_geometry::Point ball = ateam_geometry::EigenToPoint(ball_location);
+  ateam_geometry::Point middle_of_our_goal = (-4.5, 0);
+  ateam_geometry::Segement block_line = ateam_geometry::Segment(ball, middle_of_our_goal);
+  // Object to generate candidate points on this line to block
+  std::vector<ateam_geometry::Point> candidate_points;
+  Random_points_on_segment_2<ateam_geometry::Point, ateam_geometry::PointCreator> linePointCreator;
+  linePointCreator line_to_goal(ball, middle_of_our_goal);
+  // Get two defenders
+  while (defenders.length() < num_defenders) {
+    candidate_points.reserve(50);
+    std::copy_n(line_to_goal, 50, std::back_inserter(candidate_points));
+    // Remove any that will cause us to be out of bounds
+    ateam_geometry::Point previous_point = ateam_geometry::Point(-100, -100);
+    for (ateam_geometry::Point candidate : candidate_points) {
+      if (is_point_in_bounds(candidate)) {
+        if (ateam_geometry::Segment(
+            previous_point,
+            candidate).squared_length() > pow(robot_diameter, 2))
+        {
+          BehaviorGoal go_to_point {
+            BehaviorGoal::Type::MoveToPoint,
+            BehaviorGoal::Priority::Required,
+            MoveParam(ateam_geometry::PointToEigen(candidate))
+          }
+          points_around_ball.push_back(go_to_point);
```

Comment:
I'm on mobile so I might be missing it, but I don't think points around ball is defined

Reply:
I copied and pasted some code here and forgot to change variable names to match. Should be correct now (using `defenders` vector)